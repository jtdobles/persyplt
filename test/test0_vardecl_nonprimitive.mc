void main(){
    // empty array
    int[] a;
    var[][] b;
    printi(leng(a));
    printi(leng(b));
    // array size being implicitly defined by the length of literals
    var[] arr = [0, 2, 4, 6];
    string[] ar = ["H", "i"];
    printi(leng(arr));
    printi(leng(ar));

    var tupl = (1, 2.0, 3); //free declaration
    (int, float, bool) c = (4, 4.5, true);
    printi(leng(tupl));
    printi(leng(c));

    struct Student {
        name: string;
        age: int;
        major: string;
    }
    var student_info = {name: "Adam", age: 15, major: "Computer Science"};
    printf("Studentâ€™s name is {}, age is {}, and the major is {}",
        student_info.name, student_info.age, student_info.major);    



}