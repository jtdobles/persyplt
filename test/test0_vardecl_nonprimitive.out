0
0
4
2
3
'Studentâ€™s name is Adam, age is 15, and the major is Computer Science'